services/download/src/main/java/app/coronawarn/server/services/download/validation/ValidFederationKeyFilter.java

```diff
@@ -1,37 +1,123 @@
+    Validation validation = downloadServiceConfig.getValidation();
+    this.keyLength = validation.getKeyLength();
+    this.allowedReportTypes = validation.getAllowedReportTypes();
+    this.minDsos = validation.getMinDsos();
+    this.maxDsos = validation.getMaxDsos();
+    this.minRollingPeriod = validation.getMinRollingPeriod();
+    this.maxRollingPeriod = validation.getMaxRollingPeriod();
+    this.minTrl = validation.getMinTrl();
+    this.maxTrl = validation.getMaxTrl();
+  }
 
   /**
    * Accepts or rejects a key based on the evaluation of the fields against permitted values.
    */
   public boolean isValid(
       app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKey federationKey) {
-    return hasValidDaysSinceOnsetOfSymptoms(federationKey);
+    return hasValidDaysSinceOnsetOfSymptoms(federationKey)
+        && hasAllowedReportType(federationKey)
+        && hasExpectedKeyLength(federationKey)
+        && hasValidStartIntervalNumber(federationKey)
+        && hasValidTransmissionRiskLevel(federationKey)
+        && hasValidRollingPeriod(federationKey);
   }
 
   private boolean hasValidDaysSinceOnsetOfSymptoms(DiagnosisKey federationKey) {
     boolean hasValidDsos = federationKey.hasDaysSinceOnsetOfSymptoms()
-        && federationKey.getDaysSinceOnsetOfSymptoms() >= -14
-        && federationKey.getDaysSinceOnsetOfSymptoms() <= 4000;
+        && federationKey.getDaysSinceOnsetOfSymptoms() >= minDsos
+        && federationKey.getDaysSinceOnsetOfSymptoms() <= maxDsos;
     if (!hasValidDsos) {
-      logger.info("Federation DiagnosisKey found with invalid 'daysSinceOnsetOfSymptoms' value {}",
+      logger.info("Filter skipped Federation DiagnosisKey with invalid 'daysSinceOnsetOfSymptoms' value {}.",
           federationKey.getDaysSinceOnsetOfSymptoms());
     }
     return hasValidDsos;
   }
+
+  private boolean hasAllowedReportType(DiagnosisKey federationKey) {
+    boolean hasAllowedReportType = allowedReportTypes.contains(federationKey.getReportType());
+    if (!hasAllowedReportType) {
+      logger.info("Filter skipped Federation DiagnosisKey with invalid 'ReportType' {}.",
+          federationKey.getReportType());
+    }
+    return hasAllowedReportType;
+  }
+
+  private boolean hasExpectedKeyLength(DiagnosisKey federationKey) {
+    boolean hasCorrectKeyLength = federationKey.getKeyData().toByteArray().length == keyLength;
+    if (!hasCorrectKeyLength) {
+      logger.info("Filter skipped Federation DiagnosisKey with invalid 'KeyData' length {}.",
+          federationKey.getKeyData().toByteArray().length);
+    }
+    return hasCorrectKeyLength;
+  }
+
+  private boolean hasValidStartIntervalNumber(DiagnosisKey federationKey) {
+    boolean hasValidStartIntervalNumber = federationKey.hasRollingStartIntervalNumber()
+        // rolling start interval number is midnight UTC
+        && federationKey.getRollingStartIntervalNumber() % maxRollingPeriod == 0;
```

Comment:
I suggest you extract this into another method and remove the comment for clarify. For example:
```java
 && rollingStartIntervalIsMidnight(federationKey);
```